--- a/src/linguist/shared/runqttool.cpp
+++ b/src/linguist/shared/runqttool.cpp
@@ -17,9 +17,21 @@
 #include <sys/wait.h>
 #endif
 
+static QString qtBasePath(QLibraryInfo::LibraryPath location)
+{
+  switch (location) {
+    case QLibraryInfo::BinariesPath:
+      return QLatin1String(NIX_OUTPUT_OUT) + "/bin"_L1;
+    case QLibraryInfo::LibraryExecutablesPath:
+      return QLatin1String(NIX_OUTPUT_OUT) + "/libexec"_L1;
+    default:
+      return QLibraryInfo::path(location);
+  }
+}
+
 static QString qtToolFilePath(const QString &toolName, QLibraryInfo::LibraryPath location)
 {
-    QString filePath = QLibraryInfo::path(location) + u'/' + toolName;
+    QString filePath = qtBasePath(location) + u'/' + toolName;
 #ifdef Q_OS_WIN
     filePath.append(".exe"_L1);
 #endif
