--- a/src/qml/qml/qqmltypeloader.cpp
+++ b/src/qml/qml/qqmltypeloader.cpp
@@ -1248,6 +1248,7 @@ QQmlTypeLoader::QQmlTypeLoader(QV4::ExecutionEngine *engine)
     // env import paths
     addEnvImportPath("QML_IMPORT_PATH");
     addEnvImportPath("QML2_IMPORT_PATH");
+    addEnvImportPath("NIXPKGS_QT6_QML_IMPORT_PATH");
 
     addImportPath(QStringLiteral("qrc:/qt/qml"));
     addImportPath(QStringLiteral("qrc:/qt-project.org/imports"));
